codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderGenerator.kt

```diff
@@ -351,9 +351,21 @@ class ServerBuilderGenerator(
                     if (!constrainedTypeHoldsFinalType(member)) varExpr = "($varExpr).into()"
 
                     if (wrapInMaybeConstrained) {
-                        conditionalBlock("input.map(##[allow(clippy::redundant_closure)] |v| ", ")", conditional = symbol.isOptional()) {
-                            conditionalBlock("Box::new(", ")", conditional = hasBox) {
-                                rust("$maybeConstrainedVariant($varExpr)")
+                        // Temporary fix for `#[allow(clippy::redundant_closure)]` not working in `rustc` version 1.82.
```

Comment:
I have created an issue for the clippy team
https://github.com/rust-lang/rust-clippy/issues/14789

Can you update the comment to include
`TODO(https://github.com/rust-lang/rust-clippy/issues/14789) ...`